src/extension/wps/src/main/java/au/org/emii/ncdfgenerator/NcdfEncoder.java

```diff
@@ -212,38 +212,46 @@ else if (value instanceof Array) {
         // finish the file
         writer.close();
 
+        String replacedSql = markupSql(instanceId, whereClause, orderClause, definition.getFilenameTemplate().getSql());
         // get filename
-        Object filename = evaluateSql(instanceId, whereClause, orderClause, definition.getFilenameTemplate().getSql());
+        Object filename = evaluateSql(replacedSql);
 
-        // this is awful...
-        // format the file into the output stream
-        InputStream is = createWritable.getStream();
-        outputFormatter.write((String)filename, is);
-        is.close();
+        try (InputStream is = createWritable.getStream()) {
+            outputFormatter.write((String) filename, is);
+            is.close();
```

Comment:
Bugger - just noticed this line is still here.  Its now done implicitly by the try when exiting the try block and is not required (although it won't cause any issues).   I'll remove in my changes